ado_asana_sync/sync/sync.py

```diff
@@ -32,15 +34,22 @@
 
 def start_sync(app: App) -> None:
     while True:
-        app.asana_tag_gid = create_tag_if_not_existing(
-            app, get_asana_workspace(app, app.asana_workspace_name), app.asana_tag_name
-        )
-        projects = read_projects()
-        for project in projects:
-            sync_project(app, project)
+        with _TRACER.start_as_current_span("start_sync") as span:
+            span.add_event("Start sync run")
+            app.asana_tag_gid = create_tag_if_not_existing(
+                app,
+                get_asana_workspace(app, app.asana_workspace_name),
+                app.asana_tag_name,
+            )
+            projects = read_projects()
+            for project in projects:
+                sync_project(app, project)
 
-        _LOGGER.info("Sync process complete, sleeping for %s seconds", app.sleep_time)
-        sleep(app.sleep_time)
+            _LOGGER.info(
+                "Sync process complete, sleeping for %s seconds", app.sleep_time
+            )
+            span.end()
+            sleep(app.sleep_time)
```

Comment:
**Suggestion:** The `start_sync` method is quite large and complex. Consider breaking it down into smaller, more manageable methods. This would make the code easier to read and maintain.
```suggestion
# Suggestion: Break down the `start_sync` method into smaller methods
```